### PR TITLE
Chip uses new Mbed 

### PR DESCRIPTION
**Pull request by tymoteuszblochmobica**
#### Problem
Chip uses new Mbed 
Quick PR 

#### Change overview
What's in this PR

#### Testing
How was this tested? (at least one bullet point required)
* If unit tests were added, how do they cover this issue?
* If unit tests existed, how were they fixed/modified to prevent this in future?
* If new unit tests are not added, why not?
* If integration tests were added, how do they verify this change?
* If new integration tests are not added, why not?
* If manually tested, what platforms controller and device platforms were manually tested, and how?
* If no testing is required, why not?

### CODE REVIEW

**Comment by tymoteuszblochmobica** [comment]
This is initial. Im going to clean and add lockapp and shell

File: examples/lock-app/mbed/mbed_app.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
                 "CORDIO"
             ],
             "led-active-state": 1,
-            "lock-button": "USER_BUTTON",
+            "lock-button": "BUTTON1",
             "function-button": "NC"
         },
         "CY8CPROTO_062_4343W": {
```
Comment (lukgni) [inline examples/lock-app/mbed/mbed_app.json:36]: How this change is related with switching mbed to newer version?

File: examples/lock-app/mbed/mbed_app.json [hunk 2]
```diff
@@ -44,8 +44,7 @@
             "led-active-state": 1,
             "system-state-led": "LED1",
             "lock-state-led": "P9_6",
-            "lock-button": "USER_BUTTON",
-            "function-button": "P9_7"
+            "lock-button": "USER_BUTTON"
```
Comment (lukgni) [inline examples/lock-app/mbed/mbed_app.json:47]: The same: How this change is related with switching mbed to newer version?

File: examples/shell/mbed/CMakeLists.txt [hunk 3]
```diff
@@ -55,17 +53,15 @@ endif()
...
 
 if("WHD" IN_LIST MBED_TARGET_LABELS)
-    target_link_libraries(${APP_TARGET}
-        mbed-cy_psoc6_common_network mbed-emac mbed-cy_psoc6_whd
-    )
+target_link_libraries(${APP_TARGET}  mbed-cy-psoc6-common-network)
 endif()
 
-target_link_libraries(${APP_TARGET} mbed-os mbed-ble mbed-events mbed-netsocket mbed-storage mbed-storage-kv-global-api mbed-mbedtls mbed-mbedtls-cryptocell310 chip)
+target_link_libraries(${APP_TARGET} mbed-os mbed-ble mbed-events mbed-netsocket mbed-storage mbed-storage-kv-global-api mbed-mbedtls  mbed-emac chip)
```
Comment (lukgni) [inline examples/shell/mbed/CMakeLists.txt:64]: Remove extra space between `mbed-mbedtls` and `mbed-emac`

File: examples/all-clusters-app/mbed/CMakeLists.txt [hunk 4]
```diff
@@ -111,17 +109,15 @@ endif()
...
 
 if("WHD" IN_LIST MBED_TARGET_LABELS)
-    target_link_libraries(${APP_TARGET}
-        mbed-cy_psoc6_common_network mbed-emac mbed-cy_psoc6_whd
-    )
+target_link_libraries(${APP_TARGET}  mbed-cy-psoc6-common-network)
 endif()
 
-target_link_libraries(${APP_TARGET} mbed-os mbed-ble mbed-events mbed-netsocket mbed-storage mbed-storage-kv-global-api mbed-mbedtls mbed-mbedtls-cryptocell310 chip)
+target_link_libraries(${APP_TARGET} mbed-os mbed-ble mbed-events mbed-netsocket mbed-storage mbed-storage-kv-global-api mbed-mbedtls  mbed-emac chip)
```
Comment (lukgni) [inline examples/all-clusters-app/mbed/CMakeLists.txt:120]: Remove extra space between `mbed-mbedtls` and `mbed-emac`

File: examples/lighting-app/mbed/CMakeLists.txt [hunk 5]
```diff
@@ -14,7 +14,8 @@ configure_file(
     @ONLY
 )
 
-set(MBED_PATH $ENV{MBED_OS_PATH} CACHE INTERNAL "")
+#set(MBED_PATH $ENV{MBED_OS_PATH} CACHE INTERNAL "")
```
Comment (lukgni) [inline examples/lighting-app/mbed/CMakeLists.txt:17]: Can we remove this line if it's commented out?

File: examples/lighting-app/mbed/CMakeLists.txt [hunk 6]
```diff
@@ -84,17 +84,15 @@ endif()
...
 
 if("WHD" IN_LIST MBED_TARGET_LABELS)
-    target_link_libraries(${APP_TARGET}
-        mbed-cy_psoc6_common_network mbed-emac mbed-cy_psoc6_whd
-    )
+target_link_libraries(${APP_TARGET}  mbed-cy-psoc6-common-network)
 endif()
 
-target_link_libraries(${APP_TARGET} mbed-os mbed-ble mbed-events mbed-netsocket mbed-storage mbed-storage-kv-global-api mbed-mbedtls mbed-mbedtls-cryptocell310 chip)
+target_link_libraries(${APP_TARGET} mbed-os mbed-ble mbed-events mbed-netsocket mbed-storage mbed-storage-kv-global-api mbed-mbedtls  mbed-emac chip)
```
Comment (lukgni) [inline examples/lighting-app/mbed/CMakeLists.txt:95]: Remove extra space between `mbed-mbedtls` and `mbed-emac`

File: examples/lighting-app/mbed/mbed_app.json [hunk 7]
```diff
@@ -30,7 +30,7 @@
                 "CORDIO"
             ],
             "led-active-state": 1,
-            "lighting-button": "USER_BUTTON",
+            "lighting-button": "BUTTON1",
```
Comment (lukgni) [inline examples/lighting-app/mbed/mbed_app.json:33]: How this change is related with switching mbed to newer version?

File: examples/lighting-app/mbed/mbed_app.json [hunk 8]
```diff
@@ -43,8 +43,7 @@
             "led-active-state": 0,
             "system-state-led": "LED1",
             "lighting-state-led": "P9_6",
-            "lighting-button": "USER_BUTTON",
-            "function-button": "P9_7"
+            "lighting-button": "USER_BUTTON"
```
Comment (lukgni) [inline examples/lighting-app/mbed/mbed_app.json:46]: The same: How this change is related with switching mbed to newer version?

File: .vscode/launch.json [hunk 9]
```diff
@@ -170,7 +170,7 @@
             "type": "cortex-debug",
             "request": "launch",
             "cwd": "${workspaceRoot}/examples/${input:mbedApp}/mbed",
-            "executable": "./build-CY8CPROTO_062_4343W/${input:mbedFlashProfile}/chip-mbed-${input:mbedApp}-example.hex",
+            "executable": "./build-CY8CPROTO_062_4343W/${input:mbedFlashProfile}/chip-mbed-${input:mbedApp}-example.elf",
```
Comment (lukgni) [inline .vscode/launch.json:173]: Why do we change used executable format here? 
Is there any benefits by switching to .elf?
Did you compare both solutions?

File: .vscode/launch.json [hunk 10]
```diff
@@ -332,7 +332,12 @@
             "type": "pickString",
             "id": "mbedApp",
             "description": "What mbed application do you want to use?",
-            "options": ["shell", "lock-app", "lighting-app", "all-clusters-app"],
+            "options": [
+                "shell",
+                "lock-app",
+                "lighting-app",
+                "all-clusters-app"
+            ],
```
Comment (lukgni) [inline .vscode/launch.json:340]: I guess this change comes from VScode's general auto-formatting tool.
it is only present when you open our repository in vscode which is not opened in the container.
Because you didn't add anything new here, please revert this change.

**Comment by lukgni** [comment]
I didn't see any change here which modifies mbed-os version to something newer.
Have we already switched to the newer mbed-os version on development branch?

**Review by lukgni — CHANGES_REQUESTED** [review]

**Comment by lukgni** [comment]
Here is some notes (or required steps) for changing/updating submodule branch:
1. edit manually `.gitmodules` file
2. execute `git submodule sync` (only if submodule url has been changed - we need to synchronize `.gitmodules` with `.git/config`)
3. execute `git submodule update --init --remote ./third_party/mbed-os/repo` to pull changes
4. commit changes in `.gitmodules` and `./third_party/mbed-os/repo `

File: examples/lock-app/mbed/mbed_app.json (hunk 1 above)
Comment (tymoteuszblochmobica) [reply examples/lock-app/mbed/mbed_app.json:36]: They changed it in new Mbed

File: examples/lighting-app/mbed/CMakeLists.txt (hunk 5 above)
Comment (tymoteuszblochmobica) [reply examples/lighting-app/mbed/CMakeLists.txt:17]: ok done

File: examples/lighting-app/mbed/mbed_app.json (hunk 7 above)
Comment (tymoteuszblochmobica) [reply examples/lighting-app/mbed/mbed_app.json:33]: They change it in Mbed

**Review by lukgni — APPROVED** [review]

File: .gitmodules [hunk 11]
```diff
@@ -81,7 +81,7 @@
 [submodule "mbed-os"]
 	path = third_party/mbed-os/repo
 	url = https://github.com/ARMmbed/mbed-os.git
-	branch = feature-chip
+	branch = master
```
Comment (pan-) [inline .gitmodules:84]: I don't think we can use it until https://github.com/ARMmbed/mbed-os/pull/14847 has been merged

**Review by pan- — APPROVED** [review]
Approved but https://github.com/ARMmbed/mbed-os/pull/14847 must be merged first. @ATmobica can you look at the PR and address remaining comments ?

File: .gitmodules (hunk 11 above)
Comment (pan-) [reply .gitmodules:84]: It should work with the code we have now but it won't with what's present upstream.

Comment (ATmobica) [reply .gitmodules:84]: Sure, I think we can merge it to the current development but check it again before upstreaming.

**Review by pan- — APPROVED** [review]
Ok, worst case, we can extract patch master with the PR, put it in a branch and reference it from here.
